project-loader/src/main/kotlin/Loader.kt

```diff
@@ -123,7 +123,12 @@ fun <T> executeWithProject(project: File,
     
     // cleanup overridden property values to the state that they were before.
     propertyOverrides.forEach {
-        System.setProperty(it.first, it.second)
+
+        // if a property wasn't set before the value is "null"
+        // setting null as a value for a System property will result in a NPE
+        if(it.second != null) {
```

Comment:
We could also set it to `""` (empty string)  I'm not 100% sure what is better. Any opinions?